trusted_shuffler/trusted_shuffler/src/tests.rs

```diff
@@ -27,23 +30,29 @@ impl RequestHandler for TestRequestHandler {
     // A test function that processes requests.
     // In the real use-case, this function should send requests to the backend and
     // return responses.
-    async fn handle(
-        &self,
-        request: TrustedShufflerRequest,
-    ) -> anyhow::Result<TrustedShufflerResponse> {
+    async fn handle(&self, request: PlaintextRequest) -> anyhow::Result<PlaintextResponse> {
         if drop_request(&request) {
             // In that case we don't send an answer ever.
             loop {
                 tokio::task::yield_now().await;
             }
         }
-        Ok(generate_response(&request))
+        Ok(generate_plain_response(&request))
     }
 }
 
 // Non-async version of the `handle_request` function used to create expected responses.
-fn generate_response(request: &TrustedShufflerRequest) -> TrustedShufflerResponse {
-    TrustedShufflerResponse {
+fn generate_plain_response(request: &PlaintextRequest) -> PlaintextResponse {
+    PlaintextResponse {
+        headers: hyper::HeaderMap::new(),
+        data: hyper::body::Bytes::from(request.body.clone()),
+        trailers: hyper::HeaderMap::new(),
+    }
+}
+
+// Non-async version of the `handle_request` function used to create expected responses.
+fn generate_secret_response(request: &EncryptedRequest) -> EncryptedResponse {
```

Comment:
Should it be an `encrypted_response` instead of a `secret_response`?

Reply:
Done. Thanks!